smoke-tests/src/main/java/com/google/firebase/testing/CrashlyticsTest.java

```diff
@@ -0,0 +1,87 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.testing;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import android.os.Bundle;
+import androidx.test.core.app.ApplicationProvider;
+import com.google.firebase.analytics.FirebaseAnalytics;
+import com.google.firebase.crashlytics.FirebaseCrashlytics;
+
+import androidx.test.runner.AndroidJUnit4;
+import com.google.firebase.crashlytics.internal.breadcrumbs.BreadcrumbSource;
+import java.lang.reflect.Field;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+@RunWith(AndroidJUnit4.class)
+public final class CrashlyticsTest {
+
+  @Test
+  public void analyticsIntegration() {
+    // Validates that Firebase Analytics and Crashlytics interoperability is working, by confirming
+    // that events sent to Firebase Analytics are received by the Crashlytics breadcrumb handler.
+    try {
+      // need to use reflection to get the breadcrumb handler.
+      FirebaseCrashlytics crashlytics = FirebaseCrashlytics.getInstance();
+      Field coreField = crashlytics.getClass().getDeclaredField("core");
```

Comment:
I feel like this use of reflection could cause maintenance problems down the road.

There seems to be an easy workaround, given the following observation: `core` is not a private field and is visible within the `com.google.firebase.crashlytics` package. @mrichards  Can we move this test into `src/main/java/com/google/firebase/crashlytics` and access the field directly?

As for breadcrumbSource, can we make it public and annotation with `@VisibleForTesting`? (could actually be used with `core` above as well)

Reply:
@vkryachko Please take a look at the updated implementation. I moved `CrashlyticsTest` to the crashlytics package, because making `core` public in FirebaseCrashlytics makes it visible in the IDE's autocomplete for customers, even when `@VisibleForTesting` was used.

LMK if this is consistent with your suggestion.